smartsim/_core/generation/generator.py

```diff
@@ -240,57 +233,50 @@ def _symlink_files(files: t.Union[EntityFiles, None], dest: pathlib.Path) -> Non
     @staticmethod
     def _write_tagged_files(
         files: t.Union[EntityFiles, None],
-        params: t.Mapping[str, str],
+        params: t.Mapping[str, t.Any],
         dest: pathlib.Path,
     ) -> None:
         """Read, configure and write the tagged input files for
            a Job instance. This function specifically deals with the tagged
            files attached to an entity.
 
-        :param app: The Application attached to the Job
+        :param files: The paths to configure
+        :param params: A dictionary of params
         :param dest: Path to the Jobs run directory
         """
-        # Return if no files are attached
         if files is None:
             return
         if files.tagged:
             to_write = []
-
-            def _build_tagged_files(tagged: TaggedFilesHierarchy) -> None:
-                """Using a TaggedFileHierarchy, reproduce the tagged file
-                directory structure
-
-                :param tagged: a TaggedFileHierarchy to be built as a
-                               directory structure
-                """
-                for file in tagged.files:
-                    dst_path = path.join(dest, tagged.base, path.basename(file))
-                    shutil.copyfile(file, dst_path)
+            for file in files.tagged:
+                if os.path.isfile(file):
+                    dst_path = os.path.join(dest, os.path.basename(file))
+                    dst_path = shutil.copyfile(file, dst_path)
```

Comment:
Not strictly related to this ticket, but if you want, there should be no reason to manually copy the file only to then configure it place. It looks the the entrypoint is capable of making a new file on the file at the requested location.